File: Programmers_124_countries.js
function solution(n) {
    let answer = [];
    let dic = ['4','1','2']
    
    while(n > 0) {
        answer.unshift(dic[n%3]);
        n = n % 3 === 0 ? n/3 -1 : Math.floor(n/3);
    }
    
    return answer.join('');
}

File: Programmers_124_countries.cpp
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
    string answer = "";
    char dic[3] = {'4','1','2'};
    
    while(true) {
        answer = dic[n%3] + answer;
        if(n < 4) return answer;
        n = (n%3) == 0 ? (n/3) - 1 : n/3;
    }
}